tests/std/tests/P0896R4_common_iterator/test.cpp

```diff
@@ -256,12 +260,62 @@ void test_gh_2065() { // Guard against regression of GH-2065, for which we previ
         common_iterator<const int*, poor_sentinel> it2{&i};
         assert(it1 - it2 == 0);
     }
+
+    return true;
+}
+
+constexpr bool test_lwg_3574() {
+    int arr[]{11, 22, 33};
+
+    {
+        common_iterator<int*, const int*> x{arr};
+        common_iterator<int*, const int*> y{arr + 2};
+        assert(y - x == 2);
+    }
+
+    { // test that copy construction is constexpr, even when the sentinel isn't trivially copy constructible
+        common_iterator<int*, poor_sentinel> a{arr};
+        common_iterator<int*, poor_sentinel> b{a};
+        assert(b - a == 0);
+    }
+
+    common_iterator<int*, unreachable_sentinel_t> i{arr};
+    common_iterator<const int*, unreachable_sentinel_t> ci{arr + 1};
+
+    assert(*ci == 22);
+    assert(*as_const(ci) == 22);
+    assert(ci.operator->() == arr + 1);
+
+    ci = i;
+    assert(*ci == 11);
+    assert(ci == i);
+
+    assert(*++ci == 22);
+    assert(ci != i);
+
+    assert(*ci++ == 22);
+    assert(*ci == 33);
+
+    assert(iter_move(i) == 11);
+
+    common_iterator<int*, unreachable_sentinel_t> k{arr + 2};
+    iter_swap(i, k);
+    assert(arr[0] == 33);
+    assert(arr[2] == 11);
+
+    return true;
 }
 
 int main() {
     with_writable_iterators<instantiator, P>::call();
+    static_assert((with_writable_iterators<instantiator, P>::call(), true));

```

Comment:
Should we change `with_writable_iterators` to return `true` so these are all consistent?

Reply:
I've pushed one more commit to update `with_writable_iterators` usage elsewhere.